src/scaling.jl

```diff
@@ -0,0 +1,192 @@
+"""
+    Scaling <: Transform
+
+Linearly scale the data as `ax + b`, according to some statistics `a` and `b`.
+"""
+abstract type Scaling <: Transform end
+
+"""
+    IdentityScaling
+
+Represents the no-op scaling which simply returns the data it is applied on.
+"""
+struct IdentityScaling <: Scaling end
+# Convenience method
+IdentityScaling(args...; kwargs...) = IdentityScaling()
+
+apply!(A::AbstractArray{T}, scaling::IdentityScaling; kwargs...) where T <: Real = A
+
+"""
+    MeanStdScaling(mean, std) <: Scaling
+
+Linearly scale the data by a statistical `mean` and standard deviation `std`.
+The first `mean` and `std` calculated on some data will remain fixed.
+"""
+struct MeanStdScaling <: Scaling
+    # Dict keeps track of slices to protect against changes e.g. Table column order
+    # Is this going too far to generalise between Arrays and Tables? Should we use
+    # separate/parameterised structs?
+    mean::Dict{Symbol, AbstractFloat}
+    std::Dict{Symbol, AbstractFloat}
+
+    """
+        MeanStdScaling() <: Scaling
+
+    Construct a `MeanStdScaling` transform which will populate its `mean` and `std`
+    parameters from the first data it is applied to.
```

Comment:
```suggestion
    parameters from the data it is applied to in a `Dict` indexed by the array indices or 
    Table column names.
```

The idea of "first" to me reads that it applies it to the first column it comes across in the data.

I think you might also want to say that for any `MeanStdScaling` you can only populate it once.